awscli/compat.py

```diff
@@ -196,19 +182,32 @@ def _get_text_writer(stream, errors):
 
         return codecs.getwriter(encoding)(stream, errors)
 
-    def compat_open(filename, mode='r', encoding=None):
-        # See docstring for compat_open in the PY3 section above.
-        if 'b' not in mode:
-            encoding = locale.getpreferredencoding()
-        return io.open(filename, mode, encoding=encoding)
-
     def bytes_print(statement, stdout=None):
         if stdout is None:
             stdout = sys.stdout
 
         stdout.write(statement)
 
 
+def compat_open(filename, mode='r', encoding=None, access_permissions=None):
+    """Back-port open() that accepts an encoding argument.
+
+    In python3 this uses the built in open() and in python2 this
+    uses the io.open() function.
```

Comment:
Is this part still relevant given we don't support Python 2 anymore?

Reply:
No it's not. I removed it because we retained it in v2 and I wanted to minimize difference between the implementations: https://github.com/kyleknap/aws-cli/commit/692314a97d0f415e0f846ab6546b0eb6247d5394

Reply:
I don't have strong opinions though on whether to keep it.

Reply:
Yeah, I don't either. Just noted it as looking odd. We can do cleanup later.